bundles/framework/layerselector2/Flyout.js

```diff
@@ -19,18 +19,24 @@ Oskari.clazz.define('Oskari.mapframework.bundle.layerselector2.Flyout',
         this.template = null;
         this.state = null;
         this.layerTabs = [];
-        this.filterTemplate = jQuery('<div class="filter filter-border"><center><div class="filter-icon"></div><div class="filter-text"></div></center></div>');
         this.filters = [];
         this._filterNewestCount = 20;
         this._currentFilter = null;
 
         this.mapLayerService = Oskari.getSandbox().getService('Oskari.mapframework.service.MapLayerService');
         this.layerlistService = Oskari.getSandbox().getService('Oskari.mapframework.service.LayerlistService');
-
+        this.filterComponent = Oskari.clazz.create("Oskari.layerselector2.view.FilterButtons");
         this.addedButtons = {};
 
         this.layerlistService.on('Layerlist.Filter.Button.Add', function(button) {
-            me.addFilterTool(button.properties.text, button.properties.tooltip, button.properties.cls.active, button.properties.cls.deactive, button.filterId);
+            var filters = me.mapLayerService.getActiveFilters();
+            if ( filters.includes(button.filterId) || button.filterId === "newest" ) {
```

Comment:
Array.includes() doesn't work on IE so you'll have to replace it or add a polyfill for it in the core. The newest filter shouldn't require any special handling here so you should remove the handling or describe why it's needed.